File: content/4/a.go
package main

func main() {
   m := map[string]int{"year": 2019, "month": 12}

   println("example 1")
   for s := range m {
      println(s)
   }

   println("example 2")
   for s, n := range m {
      println(s, n)
   }
}

File: content/4/a.php
<?php
$m = ['year' => 2019, 'month' => 12];

echo "example 1\n";
while (true) {
   $n = current($m);
   if ($n === false) {
      break;
   }
   $s = key($m);
   echo $s, ':', $n, "\n";
   next($m);
}

echo "example 2\n";
foreach ($m as $n) {
   echo $n, "\n";
}

echo "example 3\n";
foreach ($m as $s => $n) {
   echo $s, ':', $n, "\n";
}
